### ISSUE TITLE
Pulling Windows image from ACR fails

### ISSUE THREAD

**Issue by cdietschrun** [Azure/acr#179]
**Is this a BUG REPORT or FEATURE REQUEST?**:
bug

**What happened?**:
Through Azure DevOps, ran a docker start command via Azure Pipelines on:
[internalteamnameatMicrosoft].azurecr.io/imagename:v1
and got
a Windows version 10.0.17763-based image is incompatible with a 10.0.17134 host
**What did you expect to happen?**:
The image to actually pull and run

**How do you reproduce it (as minimally and precisely as possible)?**:
Dockerfile:

FROM microsoft/dotnet-framework:4.7.2-runtime, upload to ACR, run on Windows Server Core 1803 machine in Azure.

Docker run of microsoft/dotnet-framework:4.7.2-runtime on the VM pulls and starts just fine.
**Anything else we need to know?**:

**Environment (if applicable to the issue)**:
 - Azure CLI/Powershell/SDK version whatever's in Azure DevOps agent 2.144.0
 - Docker version 18.09.1 build 20b67756d0
 - OS version Windows Server Core 1803
 - Datetime (UTC) when the issue occured
 - Registry and image names
   
   If any information is a concern to post here, you can create a [support ticket](https://azure.microsoft.com/en-us/support/create-ticket/) or send email to acrsup@microsoft.com.

**Comment by cdietschrun** [comment]
On VM: C:\Users\>systeminfo | findstr /B /C:"OS Name" /C:"OS Version"
OS Name:                   Microsoft Windows Server Datacenter
OS Version:                10.0.17134 N/A Build 17134

**Comment by mnltejaswini** [comment]
@cdietschrun  The error from docker start indicates that the host doesn't support the image version. This appears to be the windows container version compatibility issue. As suggested by the doc https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility you may have to start the container in Hyper-V isolation.

**Comment by SteveLasker** [comment]
The host limitation is a limitation of the DevOps build pool. 
You can switch to supported versions, by changing the tag
Allocate a VM
Or, use ACR Tasks where all current versions of windows conatiners are supported. Https://aka.ms/ACR/Tasks

**Comment by cdietschrun** [comment]
I've been talking with Tejaswini last night and intend to follow up with her today, but was trying to level set with exactly what I have and what I think is wrong. 

I have an Azure VM running Windows Server Core 1803 with containers (version printed above shows it's 17134). I can docker pull and run microsoft/dotnet-framework:4.7.2-runtime on that machine. I have a coworker who made a docker image from these directions https://docs.microsoft.com/en-us/visualstudio/install/build-tools-container?view=vs-2017 but uprevved to 4.7.2 because we saw OS compatibility issues with 4.7.1, which seems to make sense to me. He uploaded that to our private ACR. I then told Azure DevOps to pull this image and that's where I see the error above, which is surprising to me. Our end goal is a docker container with Windows and Visual Studio Build Tools to build and test our code against Windows in Azure DevOps CI/CD.

Steve, I'm interested in what you're saying. Are you saying this is a limitation of the hosted Azure DevOps agents? I can see that, but I'm hosting my own VM.
What tag do you mean I should switch? 
ACR tasks seem interesting but not exactly what I think we're trying to do, unless I'm missing something.

**Comment by cdietschrun** [comment]
I guess I'm missing why the DevOps pool of build agents is relevant if I'm self hosting my VM in Azure?

**Comment by SteveLasker** [comment]
Ahh, sorry. Similar problem. I believe the standard pool of VMs one Azure also don’t support the newest windows images yet. I Tejas point, if you’d run a nested VM, with the latest OS, then you can run the latest Windows container. Not pretty, I realize. This is why we’re working to avoid the need for you to manage VMs

**Comment by cdietschrun** [comment]
Thanks Steve for your input and fast replies. I acknowledge that I'm still learning all of this (our team started at Microsoft in August). Teja and I got on the phone so thank you to her! 

I think what I learned was that the 4.7.2-runtime implicitly has a multi architecture setup where it picked the host OS for the base of the image. We think this should be fixed by specifically targeting 4.7.2-runtime-windowsservercore-1803, which matches the VMs. We're building and will upload that image to confirm soon. If so, I think I understand and am hopeful. Then we can learn more about ACR tasks to see if this helps even more.

Thank you both very much for the fast help and feedback!
